### PR TITLE
[Tooling] Fix the bundler/rake failure on CI

### PR DESCRIPTION
**Pull request by AliSoftware**
This is the second attempt at https://github.com/wordpress-mobile/WordPress-Android/pull/14080 as it seems the fix wasn't sufficient and was merged too soon (before we finished testing the fix on the other PRs)
 - The first build with the fix, on the PR fixing it, passed – and PR #14080 went green.
 - But then the build failed on the other PRs I cherry-picked it on (which were triggered in parallel).
 - ~Hopefully that failure was only because the builds were racing each other as it seems like cache got picked by 2nd PR before 1st PR finished building it. So hopefully waiting until the job is done this time will help. I also used different cache keys for the different jobs just to be on the safe side.~ In fact the issue was deeper than this, see comments below and also complementary details in [the WCAndroid PR doing the same on Woo](https://github.com/woocommerce/woocommerce-android/pull/3585)

To test:
 - Ensure CI passes on this PR 
 - Then **once the CI on this PR has finished and passed**, then **cherry-pick the changes on one of the other failing PR** (#14077, #14078 and #14079)

_PR created as Draft to be sure to not merge it too fast this time_

### CODE REVIEW

**Comment by AliSoftware** [comment]
After multiple attempts and debugging, it happens that the solution is to update `bundler` to `2.2.10`.

This is because CircleCI has updated their image to use `bundler 2.2.10` apparently recently, very likely because this version contains an [important security fix](https://github.com/rubygems/rubygems/blob/master/bundler/CHANGELOG.md#security-fixes) about an issue that made some noise on internet recently.

Bundle bundler 2.2.10 also generates a different `Gemfile.lock` (to separate the gem in the lockfile by source as part of solving that security issue) and I'm guessing likely a slightly different resolution algorithm taking that into account, which is why the `Gemfile.lock` which was resolved by bundler `2.1.4` (that we were using up until now) was causing trouble to bundler `2.2.10`.

**Comment by AliSoftware** [comment]
After reading in more details [the whole PR on bundler](https://github.com/rubygems/rubygems/pull/3655/files) about this security issue, I realized that with the setup we had made bundler still use the multi-source mode, because the plugin we include in `Pluginfile` was not inside the `source` block so resolved its dependencies from a different set of sources.

<details><summary>Explanations of that side-effect issue</summary>

`bundler`'s PR explains that the whole point of fixing the security issue is to also recommend to avoid using multiple sources in Gemfiles, and using `source 'xxx' do … end` blocks to group some gems in a given source and some others (aka the ones outside the block) not having the same source will also cause issues.

In fact, as discussed in [this comment on the PR](https://github.com/rubygems/rubygems/pull/3655#issuecomment-779318620), when resolving transitive dependencies it will use the source of the parent gem, but in the case of our `Pluginfile` which doesn't specify a source, that only resolves to local gems, so wasn't able to find `rake` and `octokit` and others locally (when cache was empty at least), instead of sourcing them from rubygems.orgs like all other gems, and leading to the `Gemfile.lock` having multiple sources (hence the initially bigger diff on the `Gemfile.lock` after this change before the last commit)…
</details>

Only figured that out while trying to replicate the fix on WCAndroid in https://github.com/woocommerce/woocommerce-android/pull/3585 – so now we use a single global `source` at the top, applied to all gems including the plugin.

Pfew 😓 I hope that's it this time

**Comment by AliSoftware** [comment]
* Fix cherry-picked and being tested on Gutenberg PR too here: p1613490458125400/1613486607.116800-slack-C0126N6JH7Z
* Fix also tested on #14077, #14078

File: .circleci/config.yml [hunk 1]
```diff
@@ -249,7 +249,7 @@ jobs:
           init-submodules: true
       - checkout-submodules
       - bundle-install/bundle-install:
-          cache_key_prefix: installable-build-v2
+          cache_key_prefix: release-build-v2
```
Comment (jkmassel) [inline .circleci/config.yml:252]: Nice

File: .circleci/config.yml [hunk 2]
```diff
@@ -548,7 +548,7 @@ jobs:
           init-submodules: true
       - checkout-submodules
       - bundle-install/bundle-install:
-          cache_key_prefix: installable-build-v2
+          cache_key_prefix: translation-review-build-v2
```
Comment (jkmassel) [inline .circleci/config.yml:551]: Very nice

**Review by jkmassel — APPROVED** [review]

**Pull request merged**